### ISSUE TITLE
terraform import crash on v.2.6.3.

### ISSUE THREAD

**Issue by mskender** [Telmate/terraform-provider-proxmox#242]
Hello,

As a new user, tried to import some existing stuff from Proxmox, like this:
Defined a vm_name.tf in root:
`resource "proxmox_lxc" "vm_name" { }`

Ran `terraform import `proxmox_lxc.vm_name  proxmox_host/lxc/113`, where proxmox_host and 113 are existing Proxmox host on which CTID 113 is running. 

Got the following crash:

```
2020-11-09T11:13:11.948+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: {"data":{"ostype":"centos","rootfs":"lvm1:vm-113-disk-1,size=60G","hostname":"vm_name","digest":"7925baf0101229d1381ceca90bfb128074b2271e","swap":4096,"description":"description","cores":2,"arch":"amd64","memory":4096,"net0":"name=eth0,bridge=vmbr0,firewall=1,gw=192.168.30.254,hwaddr=AA:47:50:95:77:D6,ip=192.168.30.11/24,tag=400,type=veth"}}
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: panic: runtime error: index out of range [0] with length 0
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: goroutine 48 [running]:
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/Telmate/terraform-provider-proxmox/proxmox._resourceLxcRead(0xc000312800, 0xb45620, 0xc00007a410, 0xc609a0, 0x7fb518ae8428)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_lxc.go:635 +0x1586
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/Telmate/terraform-provider-proxmox/proxmox.resourceLxcRead(0xc000312800, 0xb45620, 0xc00007a410, 0x0, 0x0)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_lxc.go:584 +0xb4
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000318bb0, 0xdb1240, 0xc000390280, 0xc000312800, 0xb45620, 0xc00007a410, 0x0, 0x0, 0x0)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:280 +0x88
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000318bb0, 0xdb1240, 0xc000390280, 0xc00015e310, 0xb45620, 0xc00007a410, 0xc00030ca78, 0x0, 0x0, 0x0)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:554 +0x1c2
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0003081c0, 0xdb1240, 0xc000390280, 0xc0003b28a0, 0xc0003081c0, 0xc0003081d0, 0xcbcda8)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/helper/plugin/grpc_provider.go:572 +0x42f
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadResource_Handler.func1(0xdb1240, 0xc000390280, 0xc42440, 0xc0003b28a0, 0xc000390280, 0xbd1ba0, 0xd87d28, 0x0)
2020-11-09T11:13:11.950+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3276 +0x86
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve.func3.1(0xdb1300, 0xc000303020, 0xc42440, 0xc0003b28a0, 0xc000309820, 0xc000309840, 0xc00022fba0, 0x5842c8, 0xc27060, 0xc000303020)
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/plugin/serve.go:76 +0x87
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadResource_Handler(0xc5ce00, 0xc0003081c0, 0xdb1300, 0xc000303020, 0xc0003b2840, 0xc0004a0520, 0xdb1300, 0xc000303020, 0xc0003b61c0, 0x1b6)
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3278 +0x14b
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000252a80, 0xdb9dc0, 0xc00010af00, 0xc00018cf00, 0xc00007c660, 0x12db7d0, 0x0, 0x0, 0x0)
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	google.golang.org/grpc@v1.30.0/server.go:1171 +0x50a
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: google.golang.org/grpc.(*Server).handleStream(0xc000252a80, 0xdb9dc0, 0xc00010af00, 0xc00018cf00, 0x0)
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	google.golang.org/grpc@v1.30.0/server.go:1494 +0xccd
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000036300, 0xc000252a80, 0xdb9dc0, 0xc00010af00, 0xc00018cf00)
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	google.golang.org/grpc@v1.30.0/server.go:834 +0xa1
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-11-09T11:13:11.951+0100 [DEBUG] plugin.terraform-provider-proxmox_v2.6.3: 	google.golang.org/grpc@v1.30.0/server.go:832 +0x204
2020/11/09 11:13:11 [ERROR] eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/11/09 11:13:11 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/11/09 11:13:11 [TRACE] [walkImport] Exiting eval tree: import proxmox_lxc.vmname result
2020-11-09T11:13:11.952+0100 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/telmate/proxmox/2.6.3/linux_amd64/terraform-provider-proxmox_v2.6.3 pid=10143 error="exit status 2"
2020-11-09T11:13:11.952+0100 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020/11/09 11:13:11 [TRACE] vertex "import proxmox_lxc.vmname result": visit complete
2020/11/09 11:13:11 [TRACE] vertex "proxmox_lxc.vmname (import id \"proxmox_host/lxc/113\")": dynamic subgraph encountered errors
2020/11/09 11:13:11 [TRACE] vertex "proxmox_lxc.vmname (import id \"proxmox_host/lxc/113\")": visit complete
2020/11/09 11:13:11 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/telmate/proxmox\"] (close)" errored, so skipping
2020/11/09 11:13:11 [TRACE] dag/walk: upstream of "root" errored, so skipping
```

Creating and applying resources works as expected and I see that network idempotency has been fixed in the latest version. 
Haven't tried importing before, but I suspect it fails because of the following:

`panic: runtime error: index out of range [0] with length 0`
 
After it receives the LXC object description from the API.

Thanks,
Marko

**Comment by mskender** [comment]
I'm on 2.6.7 and still get panic: runtime error: index out of range [0] with length 0 trying to import. Can anybody help?

**Comment by amenome** [comment]
Getting the same experience with proxmox 2.6.7 and terraform 0.14.6.

**Comment by mskender** [comment]
The culprit is this block of code:
https://github.com/Telmate/terraform-provider-proxmox/blob/master/proxmox/resource_lxc.go
```
confRootFs := d.Get("rootfs").([]interface{})[0]
adaptedRootFs := adaptDeviceToConf(confRootFs.(map[string]interface{}), config.RootFs)
d.Set("rootfs.0", adaptedRootFs)
```
I've tried commenting it outright or making a length check:
```
	temproot := d.Get("rootfs").([]interface{})

	if len(temproot) > 0 {
		confRootFs := temproot[0]
		adaptedRootFs := adaptDeviceToConf(confRootFs.(map[string]interface{}), config.RootFs)
		d.Set("rootfs.0", adaptedRootFs)
	}
```
Making these changes and recompiling makes import run successfully but without rootfs data.
I don't know Go enough to fix it properly so it actually populates rootfs into state.

**Comment by Kusumoto** [comment]
I fixed this issue in #322 and release in version 2.6.8
